### ISSUE TITLE
Can't copy with wl-copy

### ISSUE THREAD

**Issue by caioraposo** [martanne/vis#929]
Using wl-copy results in a weird behavior where it keeps receiving text through the command prompt.
Tested with neovim and acme, it works as expected.
wl-paste works.

vis: v0.7
wlroots: v0.12
sway: v1.15

![2021-02-10T14:22:15,566869047-03:00](https://user-images.githubusercontent.com/38560726/107546922-a82c7e80-6bab-11eb-9357-472453f66ef6.png)

**Comment by martanne** [comment]
Thanks for the report.

I briefly looked into the issue and the problem is that we don't actually wait for process termination, but until all connected pipes to the standard I/O streams are closed. However, wl-copy forks to the background, but doesn't close its standard error file descriptor which means we keep waiting for it. 

This can be reproduced with the following shell pipe line which does not terminate.

```
echo "text to copy" | wl-copy 2>&1 >/dev/null | cat
```

A [similar problem](https://github.com/bugaevc/wl-clipboard/issues/100) was recently fixed for the standard input and output descriptors in a [wl-copy commit](https://github.com/bugaevc/wl-clipboard/commit/aa4633b894c3c5ae6053026529ec9288566060a6). If you apply the same changes for standard error, it should work.

It could also be argued that we should close our end of the pipe once we notice the exit of the original process.

**Comment by caioraposo** [comment]
Sorry for being a little late, but thanks for the response. Do you think we should handle possible error messages from stderr as suggested by [bugaevc](https://github.com/bugaevc/wl-clipboard/pull/110)?

**Comment by mcepl** [comment]
@ninewise, @caioraposo Is this still relevant, wasn’t this already fixed?

**Comment by caioraposo** [comment]
@mcepl I still have this issue so I put the following in my visrc:
```
vis:map(vis.modes.NORMAL, 'Y', ':> wl-copy 2>/dev/null -n<Enter>')
```

**Comment by mcepl** [comment]
> @mcepl I still have this issue so I put the following in my visrc:

1. What is your version of vis (or which commit is your vis based on)?
2. Describe exact steps how you reproduce the issue and what exactly happens when you do it?

**Comment by rnpnr** [comment]
I'm closing this issue since I believe this should have been fixed by 0cccd6e.

If its still an issue it can be reopened along with the exact steps needed to recreate it.

**Comment by fischerling** [comment]
I can still reproduce the hanging wl-copy process with the current [master](d699dea497f5ecd13cc1f3b780f84687b55e448c).

* wl-clipboard version 2.1.0
* sway version 1.8.1

## Steps to reproduce

1. Build vis on current master `./configure && make`
2. Open vis `./vis`
3. Write some text
4. Select all written lines
5. Run `:> wl-copy`. The wl-copy command does not terminate
6. Terminate the wl-copy process by hitting Ctrl+C. The selection is **NOT** copied to the clipboard
7. run `:> wl-copy 2>/dev/null`
8. The selection is properly copied and wl-copy does terminate``

**Comment by rnpnr** [comment]
I see, this makes more sense now. Right now `vis-clipboard` already redirects `xclip`'s `stderr` to `/dev/null` (see [here](https://github.com/martanne/vis/blob/master/vis-clipboard#LL109)). I'm not saying its the best solution but for the meantime I think doing the same for `wl-copy` would be sufficient. Thoughts?

**Comment by fischerling** [comment]
For me your example unfortunately still hangs even.
But I haven't investigated further.
The wl-clipboard issue tracker has a lot of open issues, which seem related.

**Comment by apprehensions** [comment]
I can still reproduce ( vis 004800e348cba1bf5f4536876b4e0eafbcf1ea58 wl-clipboard 2.2.0), and adding `2>/dev/null` fixes the problem as well .

**Comment by vaznerd** [comment]
on swaywm, foot terminal, inside tmux, using
```
> vis -v
vis 0.9 +curses +lua +tre +acl +selinux
```
on debian 13.

I have this in my config -
	vis:map(vis.modes.VISUAL, "gy", '"+y')
	vis:map(vis.modes.VISUAL_LINE, "gy", '"+y')

and when I run those, it goes to the bottom (always).

<img width="1366" height="768" alt="Image" src="https://github.com/user-attachments/assets/2a5999d6-a648-478f-a8f2-22c350d8f099" />

and somtimes clicking inside the editor(probably tmux thingy but idk) did this -
<img width="1366" height="768" alt="Image" src="https://github.com/user-attachments/assets/0c49fd16-e507-4a99-af4b-e4af441e23e2" />
basically pasted those written stuff to their. IDK whats happening.
